### PR TITLE
authn v2 support

### PR DESCRIPTION
**Pull request by pangea-andrest**

### CODE REVIEW

File: pangea-sdk/v3/service/authn/api.go [hunk 1]
```diff
@@ -0,0 +1,1419 @@
...
+
+const (
+	FTsignin FlowType = "signin"
+	FTsignup          = "signup"
+)
+
+type ProfileData map[string]string
+type Scopes []string
+
+type UserCreateRequest struct {
```
Comment (r3dcrosse) [inline pangea-sdk/v3/service/authn/api.go:155]: Should the `UserCreateRequest` struct still have all these fields
```go
type UserCreateRequest struct {
	// Base request has ConfigID for multi-config projects
	pangea.BaseRequest

	Email         string       `json:"email"`
	Authenticator string       `json:"authenticator"`
	IDProvider    IDProvider   `json:"id_provider"`
	Verified      *bool        `json:"verified,omitempty"`
	RequireMFA    *bool        `json:"require_mfa,omitempty"`
	Profile       *ProfileData `json:"profile,omitempty"`
	Scopes        *Scopes      `json:"scopes,omitempty"`
}
```

It seems like it got simplified a bit in the v2 endpoint?

Screenshot 2023-10-19 at 3 42 14 PM

File: pangea-sdk/v3/service/authn/api.go [hunk 2]
```diff
@@ -0,0 +1,1419 @@
...
+	Email         string       `json:"email"`
+	Authenticator string       `json:"authenticator"`
+	IDProvider    IDProvider   `json:"id_provider"`
+	Verified      *bool        `json:"verified,omitempty"`
+	RequireMFA    *bool        `json:"require_mfa,omitempty"`
+	Profile       *ProfileData `json:"profile,omitempty"`
+	Scopes        *Scopes      `json:"scopes,omitempty"`
+}
+
+type UserCreateResult struct {
```
Comment (r3dcrosse) [inline pangea-sdk/v3/service/authn/api.go:168]: The `UserCreateResult` struct seems like it also needs to be updated. Currently it is:

```go
type UserCreateResult struct {
	ID           string        `json:"id"`
	Email        string        `json:"email"`
	Profile      ProfileData   `json:"profile"`
	IDProviders  []string      `json:"id_providers"`
	RequireMFA   bool          `json:"require_mfa"`
	Verified     bool          `json:"verified"`
	LastLoginAt  *string       `json:"last_login_at,omitempty"`
	Disabled     bool          `json:"disabled"`
	MFAProviders []MFAProvider `json:"mfa_providers,omitempty"`
	CreatedAt    string        `json:"created_at"`
}
```

But looking at the API docs gives:

Screenshot 2023-10-19 at 3 45 58 PM

File: pangea-sdk/v3/service/authn/api.go [hunk 3]
```diff
@@ -0,0 +1,1419 @@
...
+	LastLoginAt             *string         `json:"last_login_at,omitempty"`
+	CreatedAt               string          `json:"created_at"`
+	LoginCount              int             `json:"login_count"`
+	LastLoginIP             *string         `json:"last_login_ip,omitempty"`
+	LastLoginCity           *string         `json:"last_login_city,omitempty"`
+	LastLoginCountry        *string         `json:"last_login_country,omitempty"`
+	Authenticators          []Authenticator `json:"authenticators"`
+}
+
+type UserUpdateRequest struct {
```
Comment (r3dcrosse) [inline pangea-sdk/v3/service/authn/api.go:251]:
```go
type UserUpdateRequest struct {
	// Base request has ConfigID for multi-config projects
	pangea.BaseRequest

	ID       *string `json:"id,omitempty"`
	Email    *string `json:"email,omitempty"`
	Disabled *bool   `json:"disabled,omitempty"`
}
```

Needs to be updated to:
Screenshot 2023-10-19 at 4 04 26 PM
Screenshot 2023-10-19 at 4 04 32 PM

Comment (r3dcrosse) [reply pangea-sdk/v3/service/authn/api.go:251]: Disregard, I was looking at the wrong endpoint. This is indeed correct.

**Comment by r3dcrosse** [comment]
@pangea-andrest I pushed a doc strings commit to your branch. Please pull it before committing anything else.

File: pangea-sdk/v3/service/authn/api.go (hunk 2 above)
Comment (pangea-andrest) [reply pangea-sdk/v3/service/authn/api.go:168]: Fixed!

**Pull request merged**